ls8/cpu.c

```diff
@@ -53,11 +63,31 @@ void cpu_run(struct cpu *cpu)
   {
     // TODO
     // 1. Get the value of the current instruction (in address PC).
+    unsigned char ir = cpu_ram_read(cpu, cpu->pc);
+    unsigned char operandA = cpu_ram_read(cpu, cpu->pc + 1);
+    unsigned char operandB = cpu_ram_read(cpu, cpu->pc + 2);
     // 2. Figure out how many operands this next instruction requires
     // 3. Get the appropriate value(s) of the operands following this instruction
     // 4. switch() over it to decide on a course of action.
     // 5. Do whatever the instruction should do according to the spec.
     // 6. Move the PC to the next instruction.
+    switch (ir)
+    {
+    case HLT:
+      running = 0;
+      break;
+    case LDI:
+      cpu->registers[operandA] = operandB;
+      cpu->pc += 3;
```

Comment:
you can actually determine the number of operands based on the value of ir, so you can make this drier by determining the number of operands before the switch and then increase the pc value based number of operands after the switch